app/controllers/offers/index.js

```diff
@@ -0,0 +1,37 @@
+import Controller from "@ember/controller";
+import { computed } from "@ember/object";
+import _ from "lodash";
+
+export default Controller.extend({
+  showSearch: false,
+
+  selectedSort: computed({
+    get() {
+      return this.get("sortOptions.firstObject.id");
+    },
+    set(key, value) {
+      return value.id || value;
+    }
+  }),
```

Comment:
Don't think that's a good way to use getters and setters, note that the get doesn't change based on what the setter does. And the setter doesn't set anything

**Option 1**

```suggestion
  __selectedSort: null,
  selectedSort: computed({
    get() {
      return this.get('_selectedSort') || this.get("sortOptions.firstObject.id");
    },
    set(key, value) {
      this.set('_selectedSort', value);
      return value.id || value;
    }
  }),
```

**Option 2**

Don't use a computed property at all.

```suggestion
  init() {
    this.set('selectedSort', this.get("sortOptions.firstObject.id")); // set the default value and let the user modify it later
  }
```